### PR TITLE
Week 4: Presentation

### PR DESCRIPTION
**Pull request by marcocampione**
# Assignment Proposal

## Title

Tensorboard: A suite of visualization tools to understand, debug, and optimize TensorFlow programs for ML experimentation

## Names and KTH ID
  - Marco Campione (campione@kth.se)
  - Mateus Marinheiro (mateusma@kth.se)

## Deadline

- Week 4

## Category

- Presentation

## Description

We will start the presentation with a short introduction to MLOps and its shortcomings when it comes to comparing models. Then, we will look at the Tensorboard tool and present the different features it has. We will then try to show how it can be used in the MLOps workflow and try and justify its use instead of other alternatives.

**Relevance**

MLOps needs to constantly monitor various parameters regarding machine learning models. Because of this, using a visual tool can help in this process, and this is where Tensorboard can reveal itself to be very useful.

### CODE REVIEW

**Comment by algomaster99** [comment]
Hi! Very original proposal :D I am happy to approve. But I just have one request, please make sure that you list the problems first and then explain the features that solves them. Is it okay?

**Comment by marcocampione** [comment]
Just updated the proposal, let me know if it's ok now @algomaster99

**Comment by d-melita** [comment]
# Feedback from Diogo Melita (diogogm@kth.se) and Tomás Esteves (tmbpe@kth.se)

*We certify that generative AI, incl. ChatGPT, has not been used to write this feedback. Using generative AI without permission is considered academic misconduct.*

First of all, we want to thank Mateus and Marco for allowing us to provide feedback on their presentation. Additionally, we also appreciate the kindness of both of them for providing the slides as well as the script that contained a general outline of the presentation. This helped us provide a generally better feedback as we were able to better understand the presentation.

For a reference, this feedback starts with listing high-level positive aspects and weaknesses: which are covered later on in detail (in Content and Structure/Aesthetics sections).

## High-Level Positive Aspects

- **Relevant and Attractive Topic**: The topic is very relevant for Week's 4 topic and very interesting as we were not familiar with it and it really expanded our knowledge.

- **Interesting and Engaging Content**: The flow of the presentation is very good: it introduces concepts gradually, using good visual aids to help the audience keep their focus.

- **Alternatives and Pros/Cons**: This part of the presentation is very informative. Not only it compares the technology with other ones but it also provides a clear picture of its strengths and weaknesses.

## High-Level Weaknesses

- **Lack of Expected Results in Setup**: Even though this sections helps to understand the easy setup of TensorBoard, a clear example of the expected output could be useful.

- **Hard to Understand Images**: Although we consider the supporting visual aids relevant, they are a little small which makes them hard to understand.

- **Take Away Message**: The take away message slide is to broad and should be more specific/detailed.

## Content

We really liked the TensorBoard presentation. It started with a brief explanation about MLOps before diving into TensorBoard itself. Since we had almost zero knowledge in this field, this presentation allowed us to quickly understand the core topic of it.
The slide related with some frustrations common among developers allowed us to relate with the content being presented, attracting our attention. Additionally, the tutorial/setup slide helped us understand that adding TensorBoard to an already existing project would be piece of cake. However, it could be interesting to have something to help us visualize what could be the final output. This could make it easier for people who intent to try this at home as this would allow them to not need to remember the whole presentation from the top of their heads.
Finally, we really liked the alternatives and pros/cons section. This enabled us to understand that there are more tools in the market and there's no "One Size Fits All" tool.
To finalize, after researching a more about the subject we recommend the reading of these articles. The first is a [blog](https://blogs.nvidia.com/blog/what-is-mlops/) from Nvidia explaining how they use MLOps in the industry and the second one showcases [research](https://arxiv.org/pdf/2001.07935) made in this area.

## Structure/Aesthetics

We liked the structure of the presentation. The use of images with some features to support the text and script, allowed us to visualize the evolution of the models and how we should fine tune some of their parameters. Additionally, the light elements of humour kept our attention during the whole presentation since they make the presentation smoothly and easier to follow.
Overall, the content of the presentation was great, however, how it was presented was not the best and could be perfected. The images shown along the features were a little bit small and complex for a person that never heard or used the tool. Adjusting them, would really help the audience to better understand this technology and its features.
We also found the colours on the slides to not be coherent. Some slides had dull colors while others had very vibrant ones which typically do not contrast well. We suggest choosing 3-4 colors from a palette (i.e. using [coolors.co](https://coolors.co/8ecae6-219ebc-023047-ffb703-fb8500)) as this can help mitigate this problem and allow the presentation to be better received by the audience.
Finally, the take away message slide is a bit vague and poor. Even though we understand its goal when aligned to the script, we think it would be better for this slide to have a take away message a bit more related to TensorBoard and MLOps.

## Summary

In summary, we really enjoyed how TensorBoard was presented. Its contents and structure are really interesting and engaging, which helps keep track of the whole picture. From the setup to its pros and cons, we think the presentation is really nice. We only suggest some modifications, mainly to its aesthetics, to help make the audience's eyes focused. We hope that the delivery matches the high level of the presentation. Good luck!
